src/Time.php

```diff
@@ -83,9 +84,16 @@ public function diff(self $time): \DateInterval
         return $this->toDateTime()->diff($time->toDateTime());
     }
 
+    /**
+     * @todo This should take and return DateTimeInterface on next major release
```

Comment:
Can I ask what the rationale is for waiting until the next release? `DateTimeInterface` should have many of the public methods that `DateTime` and `DateTimeImmutable` have, so it shouldn't be a breaking change.

Reply:
This decision is not mine: https://github.com/spatie/opening-hours/pull/75#issuecomment-435896626

But I agree the risk of breaking some people workflows by changing the typing (due to PHP compatibility checks) is a bit high. So it seems fair to wait for next major version, it better respects Semantic Versioning.